sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -259,6 +260,12 @@ __SYCL_TYPE(annotated_ptr) annotated_ptr<T, detail::properties_t<Props...>> {
         "constructor");
   }
 
+  // Implicit constructor was allowed for 2024.0 release. Templated to avoid
+  // ambigious/deprecation diagnostic for explicit construction.
+  template <class = void>
+  [[deprecated]] annotated_ptr(T *Ptr) noexcept
+      : m_Ptr(global_pointer_t(Ptr)) {}
+
```

Comment:
Given that we've decided not to document annotated_ptr for 2024.0, I think this part can be left out.  It will be present in the header files for 2024.0 but I don't think we have a duty to keep undocumented features working.

Reply:
Agreed, Roland please remove the deprecation warning.